src/main/java/com/exasol/exasoltestsetup/standalone/StandaloneExasolTestSetup.java

```diff
@@ -51,12 +52,47 @@ public StandaloneExasolTestSetup(final ConnectionDetailProvider connectionDetail
         exaOperationGateway.setBucketFsPort("bfsdefault", BUCKET_FS_PORT);
         this.bucketFsReadPassword = generatePassword();
         this.bucketFsWritePassword = generatePassword();
-        exaOperationGateway.setBucketPasswords(this.bucketFsReadPassword, this.bucketFsWritePassword);
-        waitForExasolToSyncTheSettings();
-        waitUntil(this::isSqlInterfaceAvailable);
+        setBucketsPasswordWithWorkaround(exaOperationGateway);
+        waitUntil(this::isSqlInterfaceAvailable, 120, "SQL interface");
         cleanTheDatabase();
     }
 
+    /**
+     * The changing of the password fails from time to time without an error (with just ni change). As a workaround we
+     * try it multiple times and check if it worked.
+     * 
+     * @param exaOperationGateway ExaOperation
+     */
+    private void setBucketsPasswordWithWorkaround(final ExaOperationGateway exaOperationGateway) {
+        int tryCounter = 0;
+        do {
+            exaOperationGateway.setBucketPasswords(this.bucketFsReadPassword, this.bucketFsWritePassword);
+            waitFor(100);
+            tryCounter++;
+            if (tryCounter > 10) {
+                throw new IllegalStateException(
+                        ExaError.messageBuilder("E-ETAJ-32").message("Failed to set BucketFS password.").toString());
+            }
+        } while (!this.isBucketFsIsAvailable());
```

Comment:
You could use a for-loop here:
for (int idxTrials=0; idxTrials < 10; idxTrials) {
...
}

Has the advantage that if avoids possible endless loops if anybody changes the code in the future...but you said it eventually will be rewritten in Go, so it's fine to keep as is....